js/client-battle.js

```diff
@@ -396,14 +396,20 @@
 						if (move.id === 'Struggle' || move.id === 'Recharge') pp = '&ndash;';
 						if (move.id === 'Recharge') move.type = '&ndash;';
 						if (name.substr(0, 12) === 'Hidden Power') name = 'Hidden Power';
+						var moveType = move.type;
+						// Update Judgment's type to match that of the Plate type, if any.
+						var item = Tools.getItem(switchables[pos].item);
+						if (move.name === 'Judgment' && item.onPlate) {
+							moveType = item.onPlate;
+						}
 						if (moveData.disabled) {
 							movebuttons += '<button disabled="disabled"' + this.tooltipAttrs(moveData.move, 'move') + '>';
 							hasDisabled = true;
 						} else {
-							movebuttons += '<button class="type-' + move.type + '" name="chooseMove" value="' + Tools.escapeHTML(moveData.move) + '"' + this.tooltipAttrs(moveData.move, 'move') + '>';
+							movebuttons += '<button class="type-' + moveType + '" name="chooseMove" value="' + Tools.escapeHTML(moveData.move) + '"' + this.tooltipAttrs(moveData.move, 'move') + '>';
 							hasMoves = true;
 						}
-						movebuttons += name + '<br /><small class="type">' + move.type + '</small> <small class="pp">' + pp + '</small>&nbsp;</button> ';
+						movebuttons += name + '<br /><small class="type">' + moveType + '</small> <small class="pp">' + pp + '</small>&nbsp;</button> ';
```

Comment:
Oh, yeah, this change makes it necessary to re-do this for mouseover as moveType is local. Let's handle it as well.